From 41bcad7889bdb4f6ce7edf38594ad1cc3e1f64ad Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 2 Mar 2025 08:22:39 +1100
Subject: [PATCH 11/15] reduce buildtests_cxx target to end2end tests

as of v1.69, the whole set of tests takes extremely long to build (>6h),
so reduce it to only "end2end" tests. This has the advantage of being
easily greppable (to make a choice of subset, and an easy rule for rebasing),
as well as being the most high-level tests that assure the library functions
as intended.

Plus four support libraries (`grpc{,++}_test{,_config,_util}`) and the vendored gtest
---
 CMakeLists.txt | 738 ++++---------------------------------------------
 1 file changed, 50 insertions(+), 688 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6b263e764..31d10c10a6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,11 +47,12 @@ set(gRPC_INSTALL_SHAREDIR "share/grpc" CACHE STRING "Installation directory for
 set(gRPC_BUILD_MSVC_MP_COUNT 0 CACHE STRING "The maximum number of processes for MSVC /MP option")
 
 # Options
+option(gRPC_CF_TESTS "Build tests" OFF)
 option(gRPC_BUILD_TESTS "Build tests" OFF)
 option(gRPC_BUILD_CODEGEN "Build codegen" ON)
 option(gRPC_DOWNLOAD_ARCHIVES "Download archives for empty 3rd party directories" ON)
 
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_CF_TESTS)
   enable_testing()
 endif()
 
@@ -90,7 +91,7 @@ set_property(CACHE gRPC_SSL_PROVIDER PROPERTY STRINGS "module" "package")
 set(gRPC_PROTOBUF_PROVIDER "module" CACHE STRING "Provider of protobuf library")
 set_property(CACHE gRPC_PROTOBUF_PROVIDER PROPERTY STRINGS "module" "package")
 
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_CF_TESTS)
   set(gRPC_BENCHMARK_PROVIDER "module" CACHE STRING "Provider of benchmark library")
   set_property(CACHE gRPC_BENCHMARK_PROVIDER PROPERTY STRINGS "module" "package")
 else()
@@ -725,7 +726,7 @@ protobuf_generate_grpc_cpp_with_import_path_correction(
   src/proto/grpc/reflection/v1alpha/reflection.proto src/proto/grpc/reflection/v1alpha/reflection.proto
 )
 
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_CF_TESTS)
 
 protobuf_generate_grpc_cpp_with_import_path_correction(
   src/proto/grpc/testing/benchmark_service.proto src/proto/grpc/testing/benchmark_service.proto
@@ -802,7 +803,7 @@ protobuf_generate_grpc_cpp_with_import_path_correction(
 
 # This enables CMake to build the project without requiring submodules
 # in the third_party directory as long as test builds are disabled.
-if (gRPC_BUILD_TESTS)
+if (gRPC_BUILD_TESTS OR gRPC_CF_TESTS)
 protobuf_generate_grpc_cpp_with_import_path_correction(
   third_party/envoy-api/envoy/admin/v3/certs.proto envoy/admin/v3/certs.proto
 )
@@ -1268,681 +1269,50 @@ protobuf_generate_grpc_cpp_with_import_path_correction(
 endif()
 
   add_custom_target(buildtests_c)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_c fd_conservation_posix_test)
-  endif()
-  add_dependencies(buildtests_c multiple_server_queues_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX OR _gRPC_PLATFORM_WINDOWS)
-    add_dependencies(buildtests_c pollset_windows_starvation_test)
-  endif()
 
   add_custom_target(buildtests_cxx)
-  add_dependencies(buildtests_cxx activity_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx address_sorting_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx address_sorting_test_unsecure)
-  endif()
   add_dependencies(buildtests_cxx admin_services_end2end_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx alarm_test)
-  endif()
-  add_dependencies(buildtests_cxx all_ok_test)
-  add_dependencies(buildtests_cxx alloc_test)
-  add_dependencies(buildtests_cxx alpn_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx alts_concurrent_connectivity_test)
-  endif()
-  add_dependencies(buildtests_cxx alts_counter_test)
-  add_dependencies(buildtests_cxx alts_crypt_test)
-  add_dependencies(buildtests_cxx alts_crypter_test)
-  add_dependencies(buildtests_cxx alts_frame_protector_test)
-  add_dependencies(buildtests_cxx alts_grpc_record_protocol_test)
-  add_dependencies(buildtests_cxx alts_handshaker_client_test)
-  add_dependencies(buildtests_cxx alts_iovec_record_protocol_test)
-  add_dependencies(buildtests_cxx alts_security_connector_test)
-  add_dependencies(buildtests_cxx alts_tsi_handshaker_test)
-  add_dependencies(buildtests_cxx alts_tsi_utils_test)
-  add_dependencies(buildtests_cxx alts_util_test)
-  add_dependencies(buildtests_cxx alts_zero_copy_grpc_protector_test)
-  add_dependencies(buildtests_cxx arena_promise_test)
-  add_dependencies(buildtests_cxx arena_test)
   add_dependencies(buildtests_cxx async_end2end_test)
-  add_dependencies(buildtests_cxx auth_context_test)
-  add_dependencies(buildtests_cxx auth_property_iterator_test)
-  add_dependencies(buildtests_cxx authorization_matchers_test)
-  add_dependencies(buildtests_cxx authorization_policy_provider_test)
-  add_dependencies(buildtests_cxx avl_test)
-  add_dependencies(buildtests_cxx aws_request_signer_test)
-  add_dependencies(buildtests_cxx backend_metrics_lb_policy_test)
-  add_dependencies(buildtests_cxx backoff_test)
-  add_dependencies(buildtests_cxx bad_ping_test)
-  add_dependencies(buildtests_cxx bad_server_response_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx bad_ssl_alpn_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx bad_ssl_cert_test)
-  endif()
-  add_dependencies(buildtests_cxx bad_streaming_id_bad_client_test)
-  add_dependencies(buildtests_cxx badreq_bad_client_test)
-  add_dependencies(buildtests_cxx basic_work_queue_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx bdp_estimator_test)
-  endif()
-  add_dependencies(buildtests_cxx bin_decoder_test)
-  add_dependencies(buildtests_cxx bin_encoder_test)
-  add_dependencies(buildtests_cxx binary_metadata_test)
-  add_dependencies(buildtests_cxx bitset_test)
-  add_dependencies(buildtests_cxx blackboard_test)
-  add_dependencies(buildtests_cxx buffer_list_test)
-  add_dependencies(buildtests_cxx byte_buffer_test)
-  add_dependencies(buildtests_cxx c_slice_buffer_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx call_arena_allocator_test)
-  endif()
-  add_dependencies(buildtests_cxx call_creds_test)
-  add_dependencies(buildtests_cxx call_filters_test)
-  add_dependencies(buildtests_cxx call_finalization_test)
-  add_dependencies(buildtests_cxx call_host_override_test)
-  add_dependencies(buildtests_cxx call_state_test)
-  add_dependencies(buildtests_cxx call_tracer_test)
-  add_dependencies(buildtests_cxx call_utils_test)
-  add_dependencies(buildtests_cxx cancel_after_accept_test)
-  add_dependencies(buildtests_cxx cancel_after_client_done_test)
-  add_dependencies(buildtests_cxx cancel_after_invoke_test)
-  add_dependencies(buildtests_cxx cancel_after_round_trip_test)
-  add_dependencies(buildtests_cxx cancel_ares_query_test)
-  add_dependencies(buildtests_cxx cancel_before_invoke_test)
-  add_dependencies(buildtests_cxx cancel_callback_test)
-  add_dependencies(buildtests_cxx cancel_in_a_vacuum_test)
-  add_dependencies(buildtests_cxx cancel_with_status_test)
-  add_dependencies(buildtests_cxx cel_authorization_engine_test)
-  add_dependencies(buildtests_cxx certificate_provider_registry_test)
-  add_dependencies(buildtests_cxx certificate_provider_store_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx cf_engine_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx cf_event_engine_test)
-  endif()
-  add_dependencies(buildtests_cxx cfstream_test)
-  add_dependencies(buildtests_cxx channel_args_test)
-  add_dependencies(buildtests_cxx channel_arguments_test)
-  add_dependencies(buildtests_cxx channel_creds_registry_test)
-  add_dependencies(buildtests_cxx channel_init_test)
-  add_dependencies(buildtests_cxx channel_stack_builder_test)
-  add_dependencies(buildtests_cxx channel_stack_test)
-  add_dependencies(buildtests_cxx channel_trace_test)
-  add_dependencies(buildtests_cxx channelz_registry_test)
-  add_dependencies(buildtests_cxx channelz_service_test)
-  add_dependencies(buildtests_cxx chaotic_good_test)
-  add_dependencies(buildtests_cxx check_gcp_environment_linux_test)
-  add_dependencies(buildtests_cxx check_gcp_environment_windows_test)
-  add_dependencies(buildtests_cxx chttp2_server_listener_test)
-  add_dependencies(buildtests_cxx chunked_vector_test)
-  add_dependencies(buildtests_cxx cli_call_test)
-  add_dependencies(buildtests_cxx client_auth_filter_test)
-  add_dependencies(buildtests_cxx client_authority_filter_test)
   add_dependencies(buildtests_cxx client_callback_end2end_test)
-  add_dependencies(buildtests_cxx client_channel_service_config_test)
-  add_dependencies(buildtests_cxx client_context_test_peer_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx client_fork_test)
-  endif()
   add_dependencies(buildtests_cxx client_interceptors_end2end_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx client_lb_end2end_test)
   endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx client_ssl_test)
-  endif()
-  add_dependencies(buildtests_cxx client_streaming_test)
-  add_dependencies(buildtests_cxx cmdline_test)
-  add_dependencies(buildtests_cxx codegen_test_full)
-  add_dependencies(buildtests_cxx codegen_test_minimal)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx combiner_test)
-  endif()
-  add_dependencies(buildtests_cxx common_closures_test)
-  add_dependencies(buildtests_cxx completion_queue_threading_test)
-  add_dependencies(buildtests_cxx compressed_payload_test)
-  add_dependencies(buildtests_cxx compression_test)
-  add_dependencies(buildtests_cxx concurrent_connectivity_test)
-  add_dependencies(buildtests_cxx connection_context_test)
-  add_dependencies(buildtests_cxx connection_prefix_bad_client_test)
-  add_dependencies(buildtests_cxx connection_refused_test)
-  add_dependencies(buildtests_cxx connectivity_state_test)
-  add_dependencies(buildtests_cxx connectivity_test)
   add_dependencies(buildtests_cxx context_allocator_end2end_test)
-  add_dependencies(buildtests_cxx context_test)
-  add_dependencies(buildtests_cxx core_configuration_test)
-  add_dependencies(buildtests_cxx cpp_impl_of_test)
-  add_dependencies(buildtests_cxx cpu_test)
-  add_dependencies(buildtests_cxx crl_provider_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx crl_ssl_transport_security_test)
-  endif()
-  add_dependencies(buildtests_cxx default_engine_methods_test)
-  add_dependencies(buildtests_cxx default_host_test)
-  add_dependencies(buildtests_cxx delegating_channel_test)
-  add_dependencies(buildtests_cxx destroy_grpclb_channel_with_active_connect_stress_test)
-  add_dependencies(buildtests_cxx directory_reader_test)
-  add_dependencies(buildtests_cxx disappearing_server_test)
-  add_dependencies(buildtests_cxx dns_resolver_cooldown_test)
-  add_dependencies(buildtests_cxx dns_resolver_test)
-  add_dependencies(buildtests_cxx down_cast_test)
-  add_dependencies(buildtests_cxx dual_ref_counted_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx dualstack_socket_test)
-  endif()
-  add_dependencies(buildtests_cxx dump_args_test)
-  add_dependencies(buildtests_cxx duplicate_header_bad_client_test)
-  add_dependencies(buildtests_cxx empty_batch_test)
   add_dependencies(buildtests_cxx end2end_test)
-  add_dependencies(buildtests_cxx endpoint_addresses_test)
-  add_dependencies(buildtests_cxx endpoint_config_test)
-  add_dependencies(buildtests_cxx endpoint_pair_test)
-  add_dependencies(buildtests_cxx env_test)
-  add_dependencies(buildtests_cxx error_details_test)
-  add_dependencies(buildtests_cxx error_test)
-  add_dependencies(buildtests_cxx error_utils_test)
-  add_dependencies(buildtests_cxx evaluate_args_test)
-  add_dependencies(buildtests_cxx event_engine_wakeup_scheduler_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx event_poller_posix_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx examine_stack_test)
-  endif()
-  add_dependencies(buildtests_cxx exception_test)
-  add_dependencies(buildtests_cxx exec_ctx_wakeup_scheduler_test)
-  add_dependencies(buildtests_cxx experiments_tag_test)
-  add_dependencies(buildtests_cxx experiments_test)
-  add_dependencies(buildtests_cxx factory_test)
-  add_dependencies(buildtests_cxx fake_resolver_test)
-  add_dependencies(buildtests_cxx fake_transport_security_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx fd_posix_test)
-  endif()
-  add_dependencies(buildtests_cxx file_watcher_certificate_provider_factory_test)
-  add_dependencies(buildtests_cxx filter_causes_close_test)
-  add_dependencies(buildtests_cxx filter_fusion_test)
-  add_dependencies(buildtests_cxx filter_init_fails_test)
-  add_dependencies(buildtests_cxx filter_test_test)
-  add_dependencies(buildtests_cxx filtered_metadata_test)
-  add_dependencies(buildtests_cxx flaky_network_test)
-  add_dependencies(buildtests_cxx flow_control_test)
-  add_dependencies(buildtests_cxx for_each_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx fork_test)
-  endif()
-  add_dependencies(buildtests_cxx forkable_test)
-  add_dependencies(buildtests_cxx format_request_test)
-  add_dependencies(buildtests_cxx frame_handler_test)
-  add_dependencies(buildtests_cxx frame_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx fuzzing_event_engine_test)
-  endif()
-  add_dependencies(buildtests_cxx fuzzing_event_engine_unittest)
-  add_dependencies(buildtests_cxx gcp_authentication_filter_test)
   add_dependencies(buildtests_cxx generic_end2end_test)
-  add_dependencies(buildtests_cxx glob_test)
-  add_dependencies(buildtests_cxx goaway_server_test)
-  add_dependencies(buildtests_cxx google_c2p_resolver_test)
-  add_dependencies(buildtests_cxx gpr_time_test)
-  add_dependencies(buildtests_cxx graceful_server_shutdown_test)
-  add_dependencies(buildtests_cxx graceful_shutdown_test)
-  add_dependencies(buildtests_cxx grpc_alts_credentials_options_test)
-  add_dependencies(buildtests_cxx grpc_audit_logging_test)
-  add_dependencies(buildtests_cxx grpc_authorization_engine_test)
-  add_dependencies(buildtests_cxx grpc_authorization_policy_provider_test)
   add_dependencies(buildtests_cxx grpc_authz_end2end_test)
-  add_dependencies(buildtests_cxx grpc_authz_test)
-  add_dependencies(buildtests_cxx grpc_byte_buffer_reader_test)
-  add_dependencies(buildtests_cxx grpc_cli)
-  add_dependencies(buildtests_cxx grpc_completion_queue_test)
-  add_dependencies(buildtests_cxx grpc_ipv6_loopback_available_test)
-  add_dependencies(buildtests_cxx grpc_tls_certificate_distributor_test)
-  add_dependencies(buildtests_cxx grpc_tls_certificate_provider_test)
-  add_dependencies(buildtests_cxx grpc_tls_certificate_verifier_test)
-  add_dependencies(buildtests_cxx grpc_tls_credentials_options_comparator_test)
-  add_dependencies(buildtests_cxx grpc_tls_credentials_options_test)
-  add_dependencies(buildtests_cxx grpc_tls_crl_provider_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx grpc_tool_test)
-  endif()
-  add_dependencies(buildtests_cxx grpclb_api_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx grpclb_end2end_test)
   endif()
-  add_dependencies(buildtests_cxx h2_ssl_cert_test)
-  add_dependencies(buildtests_cxx h2_ssl_session_reuse_test)
-  add_dependencies(buildtests_cxx h2_tls_peer_property_external_verifier_test)
-  add_dependencies(buildtests_cxx handle_tests)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx handshake_server_with_readahead_handshaker_test)
-  endif()
-  add_dependencies(buildtests_cxx head_of_line_blocking_bad_client_test)
-  add_dependencies(buildtests_cxx headers_bad_client_test)
   add_dependencies(buildtests_cxx health_service_end2end_test)
-  add_dependencies(buildtests_cxx high_initial_seqno_test)
-  add_dependencies(buildtests_cxx histogram_test)
-  add_dependencies(buildtests_cxx host_port_test)
-  add_dependencies(buildtests_cxx hpack_encoder_test)
-  add_dependencies(buildtests_cxx hpack_parser_table_test)
-  add_dependencies(buildtests_cxx hpack_parser_test)
-  add_dependencies(buildtests_cxx hpack_size_test)
-  add_dependencies(buildtests_cxx http2_client)
-  add_dependencies(buildtests_cxx http2_client_transport_test)
-  add_dependencies(buildtests_cxx http2_server_transport_test)
-  add_dependencies(buildtests_cxx http2_settings_test)
-  add_dependencies(buildtests_cxx http2_stats_test)
-  add_dependencies(buildtests_cxx http_proxy_mapper_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx httpcli_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx httpscli_test)
-  endif()
   add_dependencies(buildtests_cxx hybrid_end2end_test)
-  add_dependencies(buildtests_cxx idle_filter_state_test)
-  add_dependencies(buildtests_cxx if_list_test)
-  add_dependencies(buildtests_cxx if_test)
-  add_dependencies(buildtests_cxx init_test)
-  add_dependencies(buildtests_cxx initial_settings_frame_bad_client_test)
-  add_dependencies(buildtests_cxx insecure_security_connector_test)
-  add_dependencies(buildtests_cxx inter_activity_latch_test)
-  add_dependencies(buildtests_cxx inter_activity_pipe_test)
-  add_dependencies(buildtests_cxx interception_chain_test)
-  add_dependencies(buildtests_cxx interceptor_list_test)
-  add_dependencies(buildtests_cxx interop_client)
-  add_dependencies(buildtests_cxx interop_server)
-  add_dependencies(buildtests_cxx invalid_call_argument_test)
-  add_dependencies(buildtests_cxx invoke_large_request_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX OR _gRPC_PLATFORM_WINDOWS)
-    add_dependencies(buildtests_cxx iocp_test)
-  endif()
-  add_dependencies(buildtests_cxx istio_echo_server_test)
-  add_dependencies(buildtests_cxx join_test)
-  add_dependencies(buildtests_cxx json_object_loader_test)
-  add_dependencies(buildtests_cxx json_test)
-  add_dependencies(buildtests_cxx json_token_test)
-  add_dependencies(buildtests_cxx jwt_verifier_test)
-  add_dependencies(buildtests_cxx keepalive_timeout_test)
-  add_dependencies(buildtests_cxx lame_client_test)
-  add_dependencies(buildtests_cxx large_metadata_test)
-  add_dependencies(buildtests_cxx latch_test)
-  add_dependencies(buildtests_cxx lb_get_cpu_stats_test)
-  add_dependencies(buildtests_cxx lb_load_data_store_test)
-  add_dependencies(buildtests_cxx lb_metadata_test)
-  add_dependencies(buildtests_cxx load_config_test)
-  add_dependencies(buildtests_cxx load_file_test)
-  add_dependencies(buildtests_cxx local_security_connector_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx log_too_many_open_files_test)
-  endif()
-  add_dependencies(buildtests_cxx loop_test)
-  add_dependencies(buildtests_cxx lru_cache_test)
-  add_dependencies(buildtests_cxx map_pipe_test)
-  add_dependencies(buildtests_cxx match_promise_test)
-  add_dependencies(buildtests_cxx match_test)
-  add_dependencies(buildtests_cxx matchers_test)
-  add_dependencies(buildtests_cxx max_concurrent_streams_test)
-  add_dependencies(buildtests_cxx max_connection_age_test)
-  add_dependencies(buildtests_cxx max_connection_idle_test)
-  add_dependencies(buildtests_cxx max_message_length_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx memory_quota_stress_test)
-  endif()
-  add_dependencies(buildtests_cxx memory_quota_test)
   add_dependencies(buildtests_cxx message_allocator_end2end_test)
-  add_dependencies(buildtests_cxx message_compress_test)
-  add_dependencies(buildtests_cxx message_size_service_config_test)
-  add_dependencies(buildtests_cxx metadata_map_test)
-  add_dependencies(buildtests_cxx metrics_test)
-  add_dependencies(buildtests_cxx minimal_stack_is_minimal_test)
-  add_dependencies(buildtests_cxx miscompile_with_no_unique_address_test)
-  add_dependencies(buildtests_cxx mock_stream_test)
-  add_dependencies(buildtests_cxx mock_test)
-  add_dependencies(buildtests_cxx mpsc_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx mpscq_test)
-  endif()
-  add_dependencies(buildtests_cxx negative_deadline_test)
-  add_dependencies(buildtests_cxx no_destruct_test)
-  add_dependencies(buildtests_cxx no_logging_test)
-  add_dependencies(buildtests_cxx no_op_test)
-  add_dependencies(buildtests_cxx no_server_test)
-  add_dependencies(buildtests_cxx nonblocking_test)
-  add_dependencies(buildtests_cxx notification_test)
-  add_dependencies(buildtests_cxx num_external_connectivity_watchers_test)
-  add_dependencies(buildtests_cxx observable_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx oracle_event_engine_posix_test)
-  endif()
   add_dependencies(buildtests_cxx orca_service_end2end_test)
-  add_dependencies(buildtests_cxx orca_service_test)
-  add_dependencies(buildtests_cxx orphanable_test)
-  add_dependencies(buildtests_cxx osa_distance_test)
-  add_dependencies(buildtests_cxx out_of_bounds_bad_client_test)
-  add_dependencies(buildtests_cxx outlier_detection_lb_config_parser_test)
-  add_dependencies(buildtests_cxx outlier_detection_test)
-  add_dependencies(buildtests_cxx overload_test)
-  add_dependencies(buildtests_cxx parse_address_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx parse_address_with_named_scope_id_test)
-  endif()
-  add_dependencies(buildtests_cxx parsed_metadata_test)
-  add_dependencies(buildtests_cxx parser_test)
-  add_dependencies(buildtests_cxx party_mpsc_test)
-  add_dependencies(buildtests_cxx party_test)
-  add_dependencies(buildtests_cxx payload_test)
-  add_dependencies(buildtests_cxx percent_encoding_test)
-  add_dependencies(buildtests_cxx periodic_update_test)
-  add_dependencies(buildtests_cxx pick_first_test)
-  add_dependencies(buildtests_cxx ping_abuse_policy_test)
-  add_dependencies(buildtests_cxx ping_callbacks_test)
-  add_dependencies(buildtests_cxx ping_configuration_test)
-  add_dependencies(buildtests_cxx ping_pong_streaming_test)
-  add_dependencies(buildtests_cxx ping_rate_policy_test)
-  add_dependencies(buildtests_cxx ping_test)
-  add_dependencies(buildtests_cxx pipe_test)
-  add_dependencies(buildtests_cxx poll_test)
   add_dependencies(buildtests_cxx port_sharing_end2end_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx posix_endpoint_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx posix_engine_listener_utils_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx posix_event_engine_connect_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx posix_event_engine_native_dns_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx posix_event_engine_test)
-  endif()
-  add_dependencies(buildtests_cxx pre_stop_hook_server_test)
-  add_dependencies(buildtests_cxx prioritized_race_test)
-  add_dependencies(buildtests_cxx promise_endpoint_test)
-  add_dependencies(buildtests_cxx promise_factory_test)
-  add_dependencies(buildtests_cxx promise_map_test)
-  add_dependencies(buildtests_cxx promise_mutex_test)
-  add_dependencies(buildtests_cxx promise_test)
-  add_dependencies(buildtests_cxx proto_buffer_reader_test)
-  add_dependencies(buildtests_cxx proto_buffer_writer_test)
-  add_dependencies(buildtests_cxx proto_server_reflection_test)
-  add_dependencies(buildtests_cxx proto_utils_test)
-  add_dependencies(buildtests_cxx proxy_auth_test)
-  add_dependencies(buildtests_cxx qps_json_driver)
-  add_dependencies(buildtests_cxx qps_worker)
-  add_dependencies(buildtests_cxx query_extensions_test)
-  add_dependencies(buildtests_cxx race_test)
-  add_dependencies(buildtests_cxx random_early_detection_test)
   add_dependencies(buildtests_cxx raw_end2end_test)
-  add_dependencies(buildtests_cxx rbac_service_config_parser_test)
-  add_dependencies(buildtests_cxx rbac_translator_test)
-  add_dependencies(buildtests_cxx ref_counted_ptr_test)
-  add_dependencies(buildtests_cxx ref_counted_test)
-  add_dependencies(buildtests_cxx registered_call_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx remove_stream_from_stalled_lists_test)
-  endif()
-  add_dependencies(buildtests_cxx request_buffer_test)
-  add_dependencies(buildtests_cxx request_with_flags_test)
-  add_dependencies(buildtests_cxx request_with_payload_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx resolve_address_using_ares_resolver_posix_test)
-  endif()
-  add_dependencies(buildtests_cxx resolve_address_using_ares_resolver_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx resolve_address_using_native_resolver_posix_test)
-  endif()
-  add_dependencies(buildtests_cxx resolve_address_using_native_resolver_test)
   add_dependencies(buildtests_cxx resource_quota_end2end_stress_test)
-  add_dependencies(buildtests_cxx resource_quota_server_test)
-  add_dependencies(buildtests_cxx resource_quota_test)
-  add_dependencies(buildtests_cxx retry_cancel_after_first_attempt_starts_test)
-  add_dependencies(buildtests_cxx retry_cancel_during_delay_test)
-  add_dependencies(buildtests_cxx retry_cancel_with_multiple_send_batches_test)
-  add_dependencies(buildtests_cxx retry_cancellation_test)
-  add_dependencies(buildtests_cxx retry_disabled_test)
-  add_dependencies(buildtests_cxx retry_exceeds_buffer_size_in_delay_test)
-  add_dependencies(buildtests_cxx retry_exceeds_buffer_size_in_initial_batch_test)
-  add_dependencies(buildtests_cxx retry_exceeds_buffer_size_in_subsequent_batch_test)
-  add_dependencies(buildtests_cxx retry_lb_drop_test)
-  add_dependencies(buildtests_cxx retry_lb_fail_test)
-  add_dependencies(buildtests_cxx retry_non_retriable_status_before_trailers_test)
-  add_dependencies(buildtests_cxx retry_non_retriable_status_test)
-  add_dependencies(buildtests_cxx retry_per_attempt_recv_timeout_on_last_attempt_test)
-  add_dependencies(buildtests_cxx retry_per_attempt_recv_timeout_test)
-  add_dependencies(buildtests_cxx retry_recv_initial_metadata_test)
-  add_dependencies(buildtests_cxx retry_recv_message_replay_test)
-  add_dependencies(buildtests_cxx retry_recv_message_test)
-  add_dependencies(buildtests_cxx retry_recv_trailing_metadata_error_test)
-  add_dependencies(buildtests_cxx retry_send_initial_metadata_refs_test)
-  add_dependencies(buildtests_cxx retry_send_op_fails_test)
-  add_dependencies(buildtests_cxx retry_send_recv_batch_test)
-  add_dependencies(buildtests_cxx retry_server_pushback_delay_test)
-  add_dependencies(buildtests_cxx retry_server_pushback_disabled_test)
-  add_dependencies(buildtests_cxx retry_service_config_test)
-  add_dependencies(buildtests_cxx retry_streaming_after_commit_test)
-  add_dependencies(buildtests_cxx retry_streaming_succeeds_before_replay_finished_test)
-  add_dependencies(buildtests_cxx retry_streaming_test)
-  add_dependencies(buildtests_cxx retry_test)
-  add_dependencies(buildtests_cxx retry_throttle_test)
-  add_dependencies(buildtests_cxx retry_throttled_test)
-  add_dependencies(buildtests_cxx retry_too_many_attempts_test)
-  add_dependencies(buildtests_cxx retry_transparent_goaway_test)
-  add_dependencies(buildtests_cxx retry_transparent_max_concurrent_streams_test)
-  add_dependencies(buildtests_cxx retry_transparent_not_sent_on_wire_test)
-  add_dependencies(buildtests_cxx retry_unref_before_finish_test)
-  add_dependencies(buildtests_cxx retry_unref_before_recv_test)
-  add_dependencies(buildtests_cxx ring_buffer_test)
-  add_dependencies(buildtests_cxx ring_hash_test)
   add_dependencies(buildtests_cxx rls_end2end_test)
-  add_dependencies(buildtests_cxx rls_lb_config_parser_test)
-  add_dependencies(buildtests_cxx round_robin_test)
-  add_dependencies(buildtests_cxx secure_auth_context_test)
-  add_dependencies(buildtests_cxx secure_channel_create_test)
-  add_dependencies(buildtests_cxx secure_endpoint_test)
-  add_dependencies(buildtests_cxx security_connector_test)
-  add_dependencies(buildtests_cxx seq_test)
-  add_dependencies(buildtests_cxx sequential_connectivity_test)
-  add_dependencies(buildtests_cxx server_builder_plugin_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx server_builder_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx server_builder_with_socket_mutator_test)
-  endif()
-  add_dependencies(buildtests_cxx server_call_tracer_factory_test)
-  add_dependencies(buildtests_cxx server_chttp2_test)
-  add_dependencies(buildtests_cxx server_config_selector_test)
-  add_dependencies(buildtests_cxx server_context_test_spouse_test)
-  add_dependencies(buildtests_cxx server_early_return_test)
-  add_dependencies(buildtests_cxx server_finishes_request_test)
   add_dependencies(buildtests_cxx server_interceptors_end2end_test)
-  add_dependencies(buildtests_cxx server_registered_method_bad_client_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx server_request_call_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx server_ssl_test)
-  endif()
-  add_dependencies(buildtests_cxx server_streaming_test)
-  add_dependencies(buildtests_cxx server_test)
   add_dependencies(buildtests_cxx service_config_end2end_test)
-  add_dependencies(buildtests_cxx service_config_test)
-  add_dependencies(buildtests_cxx settings_timeout_test)
-  add_dependencies(buildtests_cxx shutdown_finishes_calls_test)
-  add_dependencies(buildtests_cxx shutdown_finishes_tags_test)
-  add_dependencies(buildtests_cxx shutdown_test)
-  add_dependencies(buildtests_cxx simple_delayed_request_test)
-  add_dependencies(buildtests_cxx simple_metadata_test)
-  add_dependencies(buildtests_cxx simple_request_bad_client_test)
-  add_dependencies(buildtests_cxx simple_request_test)
-  add_dependencies(buildtests_cxx single_set_ptr_test)
-  add_dependencies(buildtests_cxx sleep_test)
-  add_dependencies(buildtests_cxx slice_string_helpers_test)
-  add_dependencies(buildtests_cxx sockaddr_resolver_test)
-  add_dependencies(buildtests_cxx sockaddr_utils_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx socket_utils_test)
-  endif()
-  add_dependencies(buildtests_cxx sorted_pack_test)
-  add_dependencies(buildtests_cxx spinlock_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx ssl_transport_security_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx ssl_transport_security_utils_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx stack_tracer_test)
-  endif()
-  add_dependencies(buildtests_cxx stat_test)
-  add_dependencies(buildtests_cxx static_stride_scheduler_test)
-  add_dependencies(buildtests_cxx stats_test)
-  add_dependencies(buildtests_cxx status_conversion_test)
-  add_dependencies(buildtests_cxx status_flag_test)
-  add_dependencies(buildtests_cxx status_helper_test)
-  add_dependencies(buildtests_cxx status_util_test)
-  add_dependencies(buildtests_cxx stream_leak_with_queued_flow_control_update_test)
-  add_dependencies(buildtests_cxx streaming_error_response_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx streaming_throughput_test)
-  endif()
-  add_dependencies(buildtests_cxx streams_not_seen_test)
-  add_dependencies(buildtests_cxx string_ref_test)
-  add_dependencies(buildtests_cxx string_test)
-  add_dependencies(buildtests_cxx subchannel_args_test)
-  add_dependencies(buildtests_cxx switch_test)
-  add_dependencies(buildtests_cxx sync_test)
-  add_dependencies(buildtests_cxx system_roots_test)
-  add_dependencies(buildtests_cxx table_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx tcp_client_posix_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx tcp_posix_socket_utils_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx tcp_posix_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx tcp_server_posix_test)
-  endif()
-  add_dependencies(buildtests_cxx tcp_socket_utils_test)
-  add_dependencies(buildtests_cxx tdigest_test)
-  add_dependencies(buildtests_cxx test_core_channelz_channelz_test)
   add_dependencies(buildtests_cxx test_core_end2end_channelz_test)
-  add_dependencies(buildtests_cxx test_core_event_engine_posix_timer_heap_test)
-  add_dependencies(buildtests_cxx test_core_event_engine_slice_buffer_test)
-  add_dependencies(buildtests_cxx test_core_iomgr_timer_heap_test)
-  add_dependencies(buildtests_cxx test_core_security_credentials_test)
-  add_dependencies(buildtests_cxx test_core_security_ssl_credentials_test)
-  add_dependencies(buildtests_cxx test_core_slice_slice_buffer_test)
-  add_dependencies(buildtests_cxx test_core_slice_slice_test)
-  add_dependencies(buildtests_cxx test_core_util_time_test)
-  add_dependencies(buildtests_cxx test_cpp_client_credentials_test)
   add_dependencies(buildtests_cxx test_cpp_end2end_ssl_credentials_test)
-  add_dependencies(buildtests_cxx test_cpp_server_credentials_test)
-  add_dependencies(buildtests_cxx test_cpp_util_slice_test)
-  add_dependencies(buildtests_cxx test_cpp_util_time_test)
-  add_dependencies(buildtests_cxx thd_test)
-  add_dependencies(buildtests_cxx thread_manager_test)
-  add_dependencies(buildtests_cxx thread_pool_test)
-  add_dependencies(buildtests_cxx thread_quota_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx thread_stress_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx thready_posix_event_engine_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx time_jump_test)
-  endif()
-  add_dependencies(buildtests_cxx time_util_test)
-  add_dependencies(buildtests_cxx timeout_before_request_call_test)
-  add_dependencies(buildtests_cxx timeout_encoding_test)
-  add_dependencies(buildtests_cxx timer_list_test)
-  add_dependencies(buildtests_cxx timer_manager_test)
-  add_dependencies(buildtests_cxx timer_test)
-  add_dependencies(buildtests_cxx tls_certificate_verifier_test)
-  add_dependencies(buildtests_cxx tls_credentials_test)
-  add_dependencies(buildtests_cxx tls_key_export_test)
-  add_dependencies(buildtests_cxx tls_security_connector_test)
-  add_dependencies(buildtests_cxx too_many_pings_test)
-  add_dependencies(buildtests_cxx trace_flags_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx traced_buffer_list_test)
-  endif()
-  add_dependencies(buildtests_cxx trailing_metadata_test)
-  add_dependencies(buildtests_cxx transport_security_common_api_test)
-  add_dependencies(buildtests_cxx transport_security_test)
-  add_dependencies(buildtests_cxx try_join_test)
-  add_dependencies(buildtests_cxx try_seq_metadata_test)
-  add_dependencies(buildtests_cxx try_seq_test)
-  add_dependencies(buildtests_cxx unique_ptr_with_bitset_test)
-  add_dependencies(buildtests_cxx unique_type_name_test)
-  add_dependencies(buildtests_cxx unknown_frame_bad_client_test)
-  add_dependencies(buildtests_cxx uri_test)
-  add_dependencies(buildtests_cxx useful_test)
-  add_dependencies(buildtests_cxx uuid_v4_test)
-  add_dependencies(buildtests_cxx validation_errors_test)
-  add_dependencies(buildtests_cxx varint_test)
-  add_dependencies(buildtests_cxx wait_for_callback_test)
-  add_dependencies(buildtests_cxx wait_for_single_owner_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx wakeup_fd_posix_test)
-  endif()
-  add_dependencies(buildtests_cxx weighted_round_robin_config_test)
-  add_dependencies(buildtests_cxx weighted_round_robin_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX OR _gRPC_PLATFORM_WINDOWS)
-    add_dependencies(buildtests_cxx win_socket_test)
-  endif()
-  add_dependencies(buildtests_cxx window_overflow_bad_client_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX OR _gRPC_PLATFORM_WINDOWS)
-    add_dependencies(buildtests_cxx windows_endpoint_test)
-  endif()
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx work_serializer_test)
-  endif()
-  add_dependencies(buildtests_cxx write_buffering_at_end_test)
-  add_dependencies(buildtests_cxx write_buffering_test)
-  add_dependencies(buildtests_cxx write_size_policy_test)
-  if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-    add_dependencies(buildtests_cxx writes_per_rpc_test)
-  endif()
-  add_dependencies(buildtests_cxx xds_audit_logger_registry_test)
-  add_dependencies(buildtests_cxx xds_bootstrap_test)
-  add_dependencies(buildtests_cxx xds_certificate_provider_test)
-  add_dependencies(buildtests_cxx xds_client_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_cluster_end2end_test)
   endif()
-  add_dependencies(buildtests_cxx xds_cluster_resource_type_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_cluster_type_end2end_test)
   endif()
-  add_dependencies(buildtests_cxx xds_common_types_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_core_end2end_test)
   endif()
   add_dependencies(buildtests_cxx xds_credentials_end2end_test)
-  add_dependencies(buildtests_cxx xds_credentials_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_csds_end2end_test)
   endif()
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_enabled_server_end2end_test)
   endif()
-  add_dependencies(buildtests_cxx xds_endpoint_resource_type_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_fallback_end2end_test)
   endif()
@@ -1955,18 +1325,12 @@ endif()
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_http_connect_end2end_test)
   endif()
-  add_dependencies(buildtests_cxx xds_http_filters_test)
-  add_dependencies(buildtests_cxx xds_lb_policy_registry_test)
-  add_dependencies(buildtests_cxx xds_listener_resource_type_test)
-  add_dependencies(buildtests_cxx xds_metadata_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_outlier_detection_end2end_test)
   endif()
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_override_host_end2end_test)
   endif()
-  add_dependencies(buildtests_cxx xds_override_host_lb_config_parser_test)
-  add_dependencies(buildtests_cxx xds_override_host_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_pick_first_end2end_test)
   endif()
@@ -1976,14 +1340,12 @@ endif()
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_rls_end2end_test)
   endif()
-  add_dependencies(buildtests_cxx xds_route_config_resource_type_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_routing_end2end_test)
   endif()
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_security_end2end_test)
   endif()
-  add_dependencies(buildtests_cxx xds_stats_watcher_test)
   if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
     add_dependencies(buildtests_cxx xds_wrr_end2end_test)
   endif()
@@ -3163,7 +2525,7 @@ if(gRPC_INSTALL)
   )
 endif()
 
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_library(grpc_test_util
   test/core/event_engine/test_init.cc
@@ -3826,7 +3188,7 @@ if(gRPC_INSTALL)
   )
 endif()
 
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_library(gtest
   third_party/googletest/googlemock/src/gmock-cardinalities.cc
@@ -4555,7 +3917,7 @@ if(gRPC_INSTALL)
 endif()
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_library(grpc++_test
   src/cpp/client/channel_test_peer.cc
@@ -4628,7 +3990,7 @@ foreach(_hdr
 endforeach()
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_library(grpc++_test_config
   test/cpp/util/test_config_cc.cc
@@ -4680,7 +4042,7 @@ target_link_libraries(grpc++_test_config
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_library(grpc++_test_util
   src/core/util/subprocess_posix.cc
@@ -5969,7 +5331,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(admin_services_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/xds/v3/csds.pb.cc
@@ -7384,7 +6746,7 @@ target_link_libraries(arena_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(async_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/health/v1/health.pb.cc
@@ -11378,7 +10740,7 @@ target_link_libraries(client_authority_filter_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(client_callback_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -11623,7 +10985,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(client_interceptors_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -11700,7 +11062,7 @@ target_link_libraries(client_interceptors_end2end_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(client_lb_end2end_test
@@ -12628,7 +11990,7 @@ target_link_libraries(connectivity_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(context_allocator_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -13726,7 +13088,7 @@ target_link_libraries(empty_batch_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/duplicate/echo_duplicate.pb.cc
@@ -15814,7 +15176,7 @@ target_link_libraries(gcp_authentication_filter_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(generic_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/duplicate/echo_duplicate.pb.cc
@@ -16384,7 +15746,7 @@ target_link_libraries(grpc_authorization_policy_provider_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(grpc_authz_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -17432,7 +16794,7 @@ target_link_libraries(grpclb_api_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(grpclb_end2end_test
@@ -17862,7 +17224,7 @@ target_link_libraries(headers_bad_client_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(health_service_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/health/v1/health.pb.cc
@@ -18771,7 +18133,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(hybrid_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/duplicate/echo_duplicate.pb.cc
@@ -21127,7 +20489,7 @@ target_link_libraries(memory_quota_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(message_allocator_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -22296,7 +21658,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(orca_service_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/xds/v3/orca_service.pb.cc
@@ -23673,7 +23035,7 @@ target_link_libraries(poll_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(port_sharing_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -25386,7 +24748,7 @@ target_link_libraries(random_early_detection_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(raw_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/duplicate/echo_duplicate.pb.cc
@@ -26242,7 +25604,7 @@ target_link_libraries(resolve_address_using_native_resolver_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(resource_quota_end2end_stress_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -29394,7 +28756,7 @@ target_link_libraries(ring_hash_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(rls_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/lookup/v1/rls.pb.cc
@@ -30428,7 +29790,7 @@ target_link_libraries(server_finishes_request_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(server_interceptors_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -30804,7 +30166,7 @@ target_link_libraries(server_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(service_config_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/duplicate/echo_duplicate.pb.cc
@@ -33168,7 +32530,7 @@ target_link_libraries(test_core_channelz_channelz_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(test_core_end2end_channelz_test
   ${_gRPC_PROTO_GENS_DIR}/src/core/ext/transport/chaotic_good/chaotic_good_frame.pb.cc
@@ -33681,7 +33043,7 @@ target_link_libraries(test_cpp_client_credentials_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(test_cpp_end2end_ssl_credentials_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -37309,7 +36671,7 @@ target_link_libraries(xds_client_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_cluster_end2end_test
@@ -38352,7 +37714,7 @@ target_link_libraries(xds_cluster_resource_type_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_cluster_type_end2end_test
@@ -39242,7 +38604,7 @@ target_link_libraries(xds_common_types_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_core_end2end_test
@@ -39773,7 +39135,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 
 add_executable(xds_credentials_end2end_test
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/testing/echo.pb.cc
@@ -39900,7 +39262,7 @@ target_link_libraries(xds_credentials_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_csds_end2end_test
@@ -40523,7 +39885,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_enabled_server_end2end_test
@@ -41368,7 +40730,7 @@ target_link_libraries(xds_endpoint_resource_type_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_fallback_end2end_test
@@ -41898,7 +41260,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_fault_injection_end2end_test
@@ -42436,7 +41798,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_gcp_authn_end2end_test
@@ -42970,7 +42332,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_http_connect_end2end_test
@@ -45171,7 +44533,7 @@ target_link_libraries(xds_metadata_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_outlier_detection_end2end_test
@@ -45709,7 +45071,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_override_host_end2end_test
@@ -46338,7 +45700,7 @@ target_link_libraries(xds_override_host_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_pick_first_end2end_test
@@ -46877,7 +46239,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_ring_hash_end2end_test
@@ -47412,7 +46774,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_rls_end2end_test
@@ -48353,7 +47715,7 @@ target_link_libraries(xds_route_config_resource_type_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_routing_end2end_test
@@ -48891,7 +48253,7 @@ if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
 endif()
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_security_end2end_test
@@ -50025,7 +49387,7 @@ target_link_libraries(xds_stats_watcher_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_CF_TESTS)
 if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
 
   add_executable(xds_wrr_end2end_test
